### PR TITLE
Issue 53608: Handle deployments without the assay module

### PR DESCRIPTION
**Pull request by labkey-jeckels**
#### Rationale
Many failures in Sample Manager Starter, which doesn't have the Assay module.

https://teamcity.labkey.org/buildConfiguration/LabKey_Trunk_Premium_ProductSuites_SampleManagerStarter_SampleManagerStarterPostgres_Batch_4_2/3644738

```
java.lang.NullPointerException: Cannot invoke "org.labkey.api.assay.AssayService.getAssayProtocolByName(org.labkey.api.data.Container, String)" because the return value of "org.labkey.api.assay.AssayService.get()" is null
    at org.labkey.experiment.api.ExperimentServiceImpl.saveProtocol(ExperimentServiceImpl.java:6472) ~[experiment-25.9-SNAPSHOT.jar:?]
    at org.labkey.experiment.api.ExperimentServiceImpl.saveProtocol(ExperimentServiceImpl.java:6450) ~[experiment-25.9-SNAPSHOT.jar:?]
    at org.labkey.experiment.api.ExperimentServiceImpl.insertSimpleProtocol(ExperimentServiceImpl.java:8486) ~[experiment-25.9-SNAPSHOT.jar:?]
    at org.labkey.experiment.api.ExperimentServiceImpl.ensureSampleDerivationProtocol(ExperimentServiceImpl.java:7657) ~[experiment-25.9-SNAPSHOT.jar:?]
```
  
#### Changes
- Check for the service before using it

#### Tasks 📍
- [x] Needs Automation

### CODE REVIEW

**Comment by labkey-jeckels** [comment]
I queued the angry suite on this FB:

https://teamcity.labkey.org/buildConfiguration/LabKey_Trunk_Premium_ProductSuites_SampleManagerStarter_SampleManagerStarterPostgres/3646942

File: experiment/src/org/labkey/experiment/api/ExperimentServiceImpl.java [hunk 1]
```diff
@@ -6469,7 +6469,7 @@ public Protocol saveProtocol(
             if (newProtocol)
             {
                 // if protocol exists, throw error
-                if (AssayService.get().getAssayProtocolByName(protocol.getContainer(), protocol.getName()) != null)
+                if (AssayService.get() != null && AssayService.get().getAssayProtocolByName(protocol.getContainer(), protocol.getName()) != null)
```
Comment (XingY) [inline experiment/src/org/labkey/experiment/api/ExperimentServiceImpl.java:6472]: I have a related PR https://github.com/LabKey/platform/pull/6979 that I ported over this null check. My PR also handles the new test failures that complains about duplicate assay design.

Comment (labkey-jeckels) [reply experiment/src/org/labkey/experiment/api/ExperimentServiceImpl.java:6472]: Great! I'll close this PR then.
